kerberos.go

```diff
@@ -0,0 +1,128 @@
+package hdfs
+
+import (
+	"log"
+	"os"
+	"os/user"
+	"strings"
+
+	"gopkg.in/jcmturner/gokrb5.v4/client"
+	"gopkg.in/jcmturner/gokrb5.v4/config"
+	"gopkg.in/jcmturner/gokrb5.v4/credentials"
+	"gopkg.in/jcmturner/gokrb5.v4/keytab"
+)
+
+const HdfsDefaultServiceName = "nn"
+const KrbDefaultCfgPath = "/etc/krb5.conf"
+const HadoopAuthCfgPath = "hadoop.security.authentication"
+
+// GetKrbClientIfRequired returns a kerberos client if the hadoop
+// configuration or the environment variables suggest one is required
+func GetKrbClientIfRequired(conf HadoopConf) *client.Client {
+	// Check if the kerberos config path has been overriden
+	var krb5Cfg = os.Getenv("HADOOP_KRB_CONF")
+
+	if krb5Cfg == "" {
+		krb5Cfg = KrbDefaultCfgPath
+	}
+
+	// Now check if the credential cache or the keytab have been manually specified
+	keytabPath := os.Getenv("HADOOP_KEYTAB")
+
+	if keytabPath != "" {
+		return getKrbClientWithKeytab(krb5Cfg, keytabPath)
```

Comment:
Can we pass `keytabPath` to the function rather than relying only on environment variable ?

Reply:
Definitely OK with the idea, but it's slightly outside of the scope of this PR -> I'd rather wait until the branch is merged and make the hdfs library more user-friendly afterwards.